tests/browser/test/basic_playground_test.js

```diff
@@ -76,16 +79,8 @@ suite('Right Clicking on Blocks', function (done) {
   test('Collapse', async function () {
     await browser.refresh();
     const block = await dragNthBlockFromFlyout(browser, 'Loops', 0, 20, 20);
-
-    const blockId = block.id;
-    let isCollapsed = await getIsCollapsed(browser, blockId);
-    chai.assert.isFalse(isCollapsed);
```

Comment:
Why not check it's not collapsed before collappsing?

(And same below for disabling.)

Reply:
I'm joining this conversation because Beka referred to it in comments on [my recent PR](https://github.com/google/blockly/pull/7284).  Personally, I don't see a lot of value in treating these as unit tests with a single assertion per test. The manual tests that they are modeled on do a sequence of actions, checking for failure at each step, in order to test that a workflow is working. While we *can* write these tests so that they are completely separated from each other, it seems to me this creates a lot of extra work and duplication of setup.

Personally, I would rather be able to describe the tests as a set of actions by a user, and have debugging involve stepping through those actions manually to find which one failed--even if that means that an individual test does not point you directly to the specific line that failed. For my right-click tests this could also take the form of doing all of the actions (collapse, expand, disable, enable, add comment, remove comment) one after another in a single test instead of doing each as a separate test with separate creation of the block under test.

Reply:
I'm not sure where this fits into formal models of testing, although in my head this is end-to-end testing.